src/Tasks/GetComAssembliesMetadata.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Versioning;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.BackEnd;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Shared.FileSystem;
+using Microsoft.Build.Tasks.AssemblyDependency;
+using Microsoft.Build.Utilities;
+
+#nullable disable
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Resolves metadata for the specified set of COM assemblies.
+    /// </summary>
+    [SupportedOSPlatform("windows")]
+    public class GetComAssembliesMetadata : TaskExtension
+    {
+        /// <summary>
+        /// Assembly paths.
+        /// </summary>
+        private string[] _assemblyPaths = Array.Empty<string>();
+
+        /// <summary>
+        /// Set of resolved assembly metadata.
+        /// </summary>
+        private ITaskItem[] _assembliesMetadata = Array.Empty<ITaskItem>();
+
+        /// <summary>
+        /// Execute the task.
+        /// </summary>
+        public override bool Execute()
+        {
+            if (!NativeMethodsShared.IsWindows)
+            {
+                Log.LogErrorWithCodeFromResources("General.TaskRequiresWindows", nameof(GetComAssembliesMetadata));
+                return false;
+            }
+
+#if FEATURE_APPDOMAIN
+
+            var assembliesMetadata = new List<ITaskItem>();
+            foreach (string assemblyPath in AssembyPaths)
+            {
+                AssemblyInformation assemblyInformation = new(assemblyPath);
+                AssemblyAttributes attributes = assemblyInformation.GetAssemblyMetadata();
+
+                if (attributes != null)
+                {
+                    assembliesMetadata.Add(SetItemMetadata(attributes));
+                }
+            }
+
+            _assembliesMetadata = assembliesMetadata.ToArray();
+#endif
+            return true;
+        }
+
+        /// <summary>
+        /// List of assembly paths.
+        /// </summary>
+        [Required]
+        public string[] AssembyPaths
+        {
+            get => _assemblyPaths;
+
+            set
+            {
+                ErrorUtilities.VerifyThrowArgumentNull(value, nameof(AssembyPaths));
+                _assemblyPaths = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets a list of resolved assembly metadata.
+        /// </summary>
+        [Output]
+        public ITaskItem[] AssembliesMetadata => _assembliesMetadata;
+
+        /// <summary>
+        /// Sets metadata on the assembly path.
+        /// </summary>
+        private TaskItem SetItemMetadata(AssemblyAttributes attributes)
```

Comment:
super-nit / perfectly fine to ignore:

```suggestion
        private TaskItem CreateItemWithMetadata(AssemblyAttributes attributes)
```